### PR TITLE
Better rules parsing errors

### PR DESCRIPTION
**Pull request by mstemm**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area engine

/area rules

**What this PR does / why we need it**:

This cleans up rule validation to pass back consistent and appropriate error messages along with meaningful context. It can be used by callers that want to validate falco rules files.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Improve rule validation (-V) output to provide consistent info ("Ok", or error string) to stdout, including appropriate context.
```

### CODE REVIEW

File: userspace/engine/lua/compiler.lua [hunk 1]
```diff
@@ -84,29 +84,35 @@ function expand_macros(ast, defs, changed)
...
          ast.right = copy_ast_obj(defs[ast.right.value].ast)
          changed = true
       end
 
-      local changed_left = expand_macros(ast.left, defs, false)
-      local changed_right = expand_macros(ast.right, defs, false)
-      return changed or changed_left or changed_right
+      local status, changed_left = expand_macros(ast.left, defs, false)
+      if status == false then
+	 return false, changed_left
```
Comment (fntlnz) [inline userspace/engine/lua/compiler.lua]: I'm not sure about the formatting of this one

File: userspace/engine/lua/compiler.lua [hunk 2]
```diff
@@ -84,29 +84,35 @@ function expand_macros(ast, defs, changed)
...
+      if status == false then
+	 return false, changed_right
+      end
+      return true, changed or changed_left or changed_right
 
    elseif ast.type == "UnaryBoolOp" then
       if (ast.argument.type == "Macro") then
          if (defs[ast.argument.value] == nil) then
-            error("Undefined macro ".. ast.argument.value .. " used in filter.")
+	    return false, "Undefined macro ".. ast.argument.value .. " used in filter."
```
Comment (fntlnz) [inline userspace/engine/lua/compiler.lua]: ditto

File: userspace/engine/lua/rule_loader.lua [hunk 3]
```diff
@@ -190,16 +255,45 @@ function load_rules(sinsp_lua_parser,
...
+   if status == false then
+      local pat = "^([%d]+):([%d]+): "
+      -- rules is actually an error string
+
+      local row = 0
+      local col = 0
+
+      row, col = string.match(rules, pat)
+      if row ~= nil and col ~= nil then
+	 rules = string.gsub(rules, pat, "")
```
Comment (fntlnz) [inline userspace/engine/lua/rule_loader.lua]: again formatting here

File: userspace/engine/lua/compiler.lua (hunk 1 above)
Comment (mstemm) [reply userspace/engine/lua/compiler.lua]: Do we have a lua formatter yet?

**Comment by mstemm** [comment]
@fntlnz , since we agreed that any lua formatting changes will occur in a separate PR, can you approve this one? I can't merge w/ requested changes.

**Comment by mstemm** [comment]
Thank you!

**Pull request merged**